File: server/Models/Keep.cs

using System.ComponentModel.DataAnnotations;

namespace keepr.Models;

public class Keep : DbItem<int>
{
  [MinLength(1), MaxLength(50)] public string Name { get; set; }
  [MinLength(1), MaxLength(1000)] public string Description { get; set; }
  [MaxLength(500), Url] public string Img { get; set; }
  public int Views { get; set; }
  public string CreatorId { get; set; }
  public Profile Creator { get; set; }
  public int Kept { get; set; }
}

File: client/src/models/Keep.js
import { DbItem } from "./DbItem.js";
import { Profile } from "./Profile.js";

export class Keep extends DbItem {
  constructor(data){
    super(data);
    this.name = data.name;
    this.description = data.description;
    this.img = data.img;
    this.views = data.views;
    this.creatorId = data.creatorId;
    this.creator = new Profile(data.creator)
    this.kept = data.kept;
  }
}